### DISCUSSION TITLE
What does the final executable consist of + Alternate rendering api

### DISCUSSION THREAD

**Discussion by zuixalias** [emilk/egui#1651] [Q&A] [answered]
The exe size of a release build on windows with no extra code (boilerplate only) is ~2.5mb.
Is it purely due to all the code or is there anything in it that can be trimmed...?

(Like, it would be nice if any feature that's not being used is not included or something like that...also, i currently assume that the size is due to the way egui is implemented because of which it has to use lot's of code to get it running/ or it includes the rendering api or something idk)

Someone please let me know if it is possible to cut down the size further any how (i am expecting ~500kb at max)

// --------------------------------------

Is it possible to use GDI for rendering?

**Comment by emilk** [comment, marked as answer]
egui/epaint comes bundled with fonts, but you can turn it off: https://github.com/emilk/egui/blob/master/egui/Cargo.toml#L38

**Reply by Nrosa01** [reply]
I have the same problem and egui cargo.toml shows that all features are disabled by default. The link you pointed to is invalid. Are the fonts still getting included nowadays?

**Comment by ShellCode33** [comment]
The `default_fonts` feature is enabled by default in eframe:

https://github.com/emilk/egui/blob/707cd03357f926d040acb241a5c42d8e91a83d8e/crates/eframe/Cargo.toml#L34-L43

as well as in egui:

https://github.com/emilk/egui/blob/707cd03357f926d040acb241a5c42d8e91a83d8e/crates/egui/Cargo.toml#L27

Disable the default features in your `Cargo.toml` as much as possible and reenable what you need:

As an exemple for future readers, here's what I have so far:

```
[dependencies]
egui = { version = "0.29.1", default-features = false }
egui_extras = { version = "0.29.1", default-features = false, features = [
  "image",
] }
eframe = { version = "0.29.1", default-features = false, features = [
  "glow",
  "wayland",
  "x11",
] }
image = { version = "0.25", default-features = false, features = ["png"] }
```

My binary size is about 7.5MiB (for ~100 LoC and no images) with debug symbols not stripped and the compilation process not optimized for size, just the cargo defaults.
